fpdf/graphics_state.py

```diff
@@ -0,0 +1,113 @@
+class GraphicsStateMixin:
+    """Mixin class for managing a stack of graphics state variables.
+
+    To the subclassing library and its users, the variables look like
+    normal instance attributes. But by the magic of properties, we can
+    push and pop levels as needed, and users will always see and modify
+    just the current version.
+
+    This class is mixed in by fpdf.FPDF(), and is not meant to be used
+    directly by user code.
+    """
+
+    def __init__(self):
+        self.__statestack = [
+            dict(
+                draw_color="0 G",
+                fill_color="0 g",
+                text_color="0 g",
+                underline=False,
+                font_style="",
+                font_stretching=100,
+                font_family="",
+                font_size_pt=0,
+                font_size=0,
+                dash_pattern="[] 0 d",
+            ),
+        ]
+
+    def _push_local_stack(self):
+        self.__statestack.append(self.__statestack[-1].copy())
+
+    def _pop_local_stack(self):
+        del self.__statestack[-1]
+
+    @property
+    def draw_color(self):
+        return self.__statestack[-1]["draw_color"]
+
+    @draw_color.setter
```

Comment:
Il faut que les "setters" de la classe parente fassent désormais appel à ces méthodes : `FPDF.set_stretching`, `FPDF.set_draw_color`, `FPDF.set_fill_color`, `FPDF.set_text_color`, `FPDF.set_line_width`, `FPDF.set_dash_pattern`, `FPDF.set_font`, `FPDF.set_font_size`

Reply:
(sorry for the sudden mix of French in the review, my mistake)

Reply:
Ok, I see now why this is not needed